roles/kubernetes/preinstall/tasks/0040-set_facts.yml

```diff
@@ -6,6 +6,14 @@
   tags:
     - facts
 
+- name: Set os_family fact for Kylin Linux Advanced Server
+  set_fact:
+    ansible_os_family: "RedHat"
```

Comment:
HI @cristicalin 

The "Kylin Linux Advanced Server Support" is reported "ansible_os_family = "RedHat" https://github.com/ansible/ansible/pull/78232/files#diff-5fc85064a2b087baf52a8e41993aa98df9cab31ea48f10450a8776373dcdc1aeR514 .

And it will release in ansible v2.14. Today we use ansible v2.12 in KubeSpray.
I think after the kubespray support ansible v2.14, the command should be removed, and it's useful in ansible v2.12.

Reply:
Hi @cristicalin,  Can we handle this issue temporarily as @yankay described I would remove the `set_fact` step when ansible is upgraded to v2.14, what do you think?

Reply:
@yankay's workaround seems good for me.
Even if keeping this workaround code at ansible v2.14+, that still should work.

One more thing is now those vagrant jobs are failed due to another similar distributions like
```
fatal: [k8s-1]: FAILED! => {"msg": "The conditional check 'ansible_distribution == \"UnionTech\"' failed. The error was: error while evaluating conditional (ansible_distribution == \"UnionTech\"): 'ansible_distribution' is undefined\n\nThe error appears to be in '/builds/kargo-ci/kubernetes-sigs-kubespray/roles/bootstrap-os/tasks/main.yml': line 87, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Set os_family fact for UOS Linux\n  ^ here\n"}
fatal: [k8s-2]: FAILED! => {"msg": "The conditional check 'ansible_distribution == \"UnionTech\"' failed. The error was: error while evaluating conditional (ansible_distribution == \"UnionTech\"): 'ansible_distribution' is undefined\n\nThe error appears to be in '/builds/kargo-ci/kubernetes-sigs-kubespray/roles/bootstrap-os/tasks/main.yml': line 87, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Set os_family fact for UOS Linux\n  ^ here\n"}
```
So this block would be necessary for UOS Linux.

Reply:
Hi, @oomichi , I'll remove the code you metioned. And I found the further reason is about the "--flush-cache" option in the CI. If we don't use this option, there will not have this error.

Reply:
Hi, @oomichi , I'll remove the code you metioned. And I found the further reason is about the "--flush-cache" option in the CI. If we don't use this option, there will not have this error.

Reply:
Thanks in advance.
I think this pull request needs to solve vagrant-* job issues before merging.

/lgtm cancel

Reply:
After merging https://github.com/kubernetes-sigs/kubespray/pull/9462 please rebase this pull request for the latest master branch to verify this solves the issue.